### ISSUE TITLE
SA1513 and SA1500 for Auto-Implemented Properties

### ISSUE THREAD

**Issue by GregReddick** [DotNetAnalyzers/StyleCopAnalyzers#1613]
I want to re-open the issue discussed in #1586. This code should be allowed without triggering SA1500 and SA1513:

public PointF A
{
	get;
	set;
} = new PointF(0, 0);

This is how I would line break and indent this. I know that many people would put this on one line:

public PointF A { get; set;	} = new PointF(0, 0);

Which should also be allowed, but not how I would do it. This is a read-write property that has a default, but no backing field. It is allowed by C# version 6, and must be supported without reporting warnings.

**Comment by GregReddick** [comment]
nuts, it removed my indenting of the get; set;.

**Comment by sharwell** [comment]
@GregReddick I fixed the indentation. :smile:

**Comment by eholman** [comment]
This is triggering the same rules:

```csharp
[field]
public string Name
{
    get;
    private set => this.SetProperty(ref field, value);
} = null!;
```

This is allowed...

```csharp
[field]
public string Name
{
    get;
    private set => this.SetProperty(ref field, value);
}

    = null!;
```

Putting them on a single line isn't allowed either.

**Comment by eholman** [comment]
The rules also fire with this setup:

```csharp
public Guid Id
{
    get;
    set
    {
        this.IdIsSet = true; // or any other additional statement
        field = value;
    }
} = Guid.NewGuid();
```

This is allowed:

```csharp
public Guid Id
{
    get;
    set
    {
        this.IdIsSet = true;
        field = value;
    }
}

    = Guid.NewGuid();
```

Putting them on one line isn't allowed
